duckduckgo/macos-browser
# Add Phishing Detection Data Update Script

Task/Issue URL: https://app.asana.com/0/1204023833050360/1208270234071172/f
Tech Design URL: https://app.asana.com/0/481882893211075/1207483114414814
CC:

**Description**:
In [✓ Tech Design: Phishing Protection Data Updates](https://app.asana.com/0/481882893211075/1207483114414814/f) we defined an approach to get embedded data for phishing protection into the app builds. This pattern was implemented, but two small components still need merging:
The update script:
A small bash script that pulls data from the API into the repo in JSON format, and updates the checksums + revision values in PhishingDetection.swift .
This has already been implemented, it just needs testing and merging:

A GitHub Action workflow that executes this script once a week and creates a PR that merges this data into the new release build.
[Example Script](https://github.com/duckduckgo/macos-browser/blob/0fb680211ad05dacb621b351a8f7b266e7239b7d/.github/workflows/update_phishing_detection_data.yml)
The secrets and workflow have already been defined, it just needs to be tested and merged.

**Steps to test this PR**:
1. Execute `bash scripts/update_phishing_detection_data.sh`
2. Ensure it succeeds
3. Check git, validate that the version has been updated to a higher version, and SHA hashes were updated in `./DuckDuckGo/PhishingDetection/PhishingDetection.swift`



**Definition of Done**:


---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)
